lib/_stream_readable.js

```diff
@@ -550,10 +550,12 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
   }
 
   src.on('data', ondata);
+  var increasedAwaitDrain = false;
```

Comment:
also we probably wanna move up the declaration of `increasedAwaitDrain` to before the `ondata` handler is attached. i'm pretty sure that might fire sync in some cases